.github/workflows/action-pypi-build-and-deploy.yml

```diff
@@ -0,0 +1,44 @@
+name: PyPi Build and Deploy 🐍📦
+
+on:
+  release:
+    types: [published]
+  # use this for testing
+  push:
+    branches:
+      - deploy_pypi
```

Comment:
```suggestion
  push:
    branches:
      - master
```

Reply:
For testing that the build succeeds, upload will only happen on tags (see line 41)